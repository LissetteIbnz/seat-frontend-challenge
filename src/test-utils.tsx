import * as React from "react";
import { fireEvent, render, RenderOptions, RenderResult } from "@testing-library/react";
import { ThemeProvider } from "./core/theme";
import { StylesProvider } from "@chakra-ui/react";

type RenderProps = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">,
) => RenderResult;

const WrapperProviders: React.FC = ({ children }) => (
  <StylesProvider value={{}}>
    <ThemeProvider>{children}</ThemeProvider>
  </StylesProvider>
);

const customRender: RenderProps = (ui, renderOptions) =>
  render(ui, { wrapper: WrapperProviders, ...renderOptions });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const changeInput = (input: HTMLInputElement | HTMLTextAreaElement, value: string) =>
  fireEvent.change(input, { target: { value } });
