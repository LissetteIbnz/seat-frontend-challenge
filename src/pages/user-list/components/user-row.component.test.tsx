import { render, screen, fireEvent } from "test-utils";
import { UserRowComponent } from "./user-row.component";

type Props = React.ComponentProps<typeof UserRowComponent>;

const renderWithTable = (props: Props) => {
  return render(
    <table>
      <tbody>
        <UserRowComponent {...props} />
      </tbody>
    </table>,
  );
};

describe("pages/user-list/components/UserRowComponent", () => {
  let props: Props;

  beforeEach(() => {
    props = {
      row: {
        id: 1,
        fullName: "John Doe",
        email: "irrelevant",
      },
      onEdit: jest.fn(),
      onDelete: jest.fn(),
    };
  });

  it("should render correctly", () => {
    const { container } = renderWithTable(props);

    expect(container).toMatchSnapshot();
  });

  it("should call onEdit when a user clicks on edit button", () => {
    renderWithTable(props);

    const editButton = screen.getByLabelText("Edit");
    fireEvent.click(editButton);

    expect(props.onEdit).toHaveBeenCalledTimes(1);
  });

  it("should call onDelete when a user clicks on delete button", () => {
    renderWithTable(props);

    const deleteButton = screen.getByLabelText("Delete");
    fireEvent.click(deleteButton);

    expect(props.onDelete).toHaveBeenCalledTimes(1);
    expect(props.onDelete).toHaveBeenCalledWith({
      id: props.row.id.toString(),
      name: props.row.fullName,
    });
  });
});
