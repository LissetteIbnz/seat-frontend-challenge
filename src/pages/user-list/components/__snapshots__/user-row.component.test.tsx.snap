// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/user-list/components/UserRowComponent should render correctly 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="css-0"
        role="row"
      >
        <td
          class="css-0"
          role="gridcell"
        >
          1
        </td>
        <td
          class="css-0"
          role="gridcell"
        >
          John Doe
        </td>
        <td
          class="css-0"
          role="gridcell"
        >
          irrelevant
        </td>
        <td
          class="css-0"
          role="gridcell"
        >
          <button
            aria-label="Edit"
            class="chakra-button css-vfkdx4"
            type="button"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="1.1rem"
              viewBox="0 0 32 32"
              width="1.1rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M 23.90625 3.96875 C 22.859286 3.96875 21.813178 4.3743215 21 5.1875 L 5.40625 20.78125 L 5.1875 21 L 5.125 21.3125 L 4.03125 26.8125 L 3.71875 28.28125 L 5.1875 27.96875 L 10.6875 26.875 L 11 26.8125 L 11.21875 26.59375 L 26.8125 11 C 28.438857 9.373643 28.438857 6.813857 26.8125 5.1875 C 25.999322 4.3743215 24.953214 3.96875 23.90625 3.96875 z M 23.90625 5.875 C 24.409286 5.875 24.919428 6.1069285 25.40625 6.59375 C 26.379893 7.567393 26.379893 8.620107 25.40625 9.59375 L 24.6875 10.28125 L 21.71875 7.3125 L 22.40625 6.59375 C 22.893072 6.1069285 23.403214 5.875 23.90625 5.875 z M 20.3125 8.71875 L 23.28125 11.6875 L 11.1875 23.78125 C 10.533142 22.500659 9.4993415 21.466858 8.21875 20.8125 L 20.3125 8.71875 z M 6.9375 22.4375 C 8.1365842 22.923393 9.0766067 23.863416 9.5625 25.0625 L 6.28125 25.71875 L 6.9375 22.4375 z"
                fill="currentColor"
                overflow="visible"
              />
            </svg>
          </button>
          <button
            aria-label="Delete"
            class="chakra-button css-azs6eb"
            type="button"
          >
            <svg
              aria-hidden="true"
              fill="currentColor"
              focusable="false"
              height="1.1rem"
              version="1.1"
              viewBox="0 0 791.908 791.908"
              width="1.1rem"
              x="0px"
              xmlns="http://www.w3.org/2000/svg"
              y="0px"
            >
              <g>
                <path
                  d="M648.587,99.881H509.156C500.276,43.486,452.761,0,394.444,0S287.696,43.486,279.731,99.881H142.315   c-26.733,0-48.43,21.789-48.43,48.43v49.437c0,24.719,17.761,44.493,41.564,47.423V727.64c0,35.613,28.655,64.268,64.268,64.268   h392.475c35.613,0,64.268-28.655,64.268-64.268V246.087c23.711-3.937,41.564-23.711,41.564-47.423v-49.437   C697.017,121.67,675.228,99.881,648.587,99.881z M394.444,36.62c38.543,0,70.219,26.733,77.085,63.261H316.351   C324.225,64.268,355.901,36.62,394.444,36.62z M618.924,728.739c0,14.831-11.901,27.648-27.648,27.648H198.71   c-14.831,0-27.648-11.901-27.648-27.648V247.185h446.948v481.554H618.924z M660.397,197.748c0,6.958-4.944,11.902-11.902,11.902   H142.223c-6.958,0-11.902-4.944-11.902-11.902v-49.437c0-6.958,4.944-11.902,11.902-11.902h505.265   c6.958,0,11.901,4.944,11.901,11.902v49.437H660.397z M253.09,661.45V349.081c0-9.887,7.873-17.761,17.761-17.761   s17.761,7.873,17.761,17.761V661.45c0,9.887-7.873,17.761-17.761,17.761C260.964,680.309,253.09,671.337,253.09,661.45z    M378.606,661.45V349.081c0-9.887,7.873-17.761,17.761-17.761c9.887,0,17.761,7.873,17.761,17.761V661.45   c0,9.887-7.873,17.761-17.761,17.761C386.57,680.309,378.606,671.337,378.606,661.45z M504.212,661.45V349.081   c0-9.887,7.873-17.761,17.761-17.761s17.761,7.873,17.761,17.761V661.45c0,9.887-7.873,17.761-17.761,17.761   C513.093,680.309,504.212,671.337,504.212,661.45z"
                />
              </g>
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
              <g />
            </svg>
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
